lib/src/dart2_9_suggestors/dart2_9_utilities.dart

```diff
@@ -0,0 +1,170 @@
+/// |--------------------------------------------------------|-------------|
+/// | ((props) {}, $FooConfig)                               | $FooConfig  |
+/// | ((props) {}, _$FooConfig)                              | _$FooConfig |
+/// | ((props) {}, UiFactoryConfig())                        | null        |
+SimpleIdentifier getGeneratedFactoryConfigArg(ArgumentList argList) {
+  final args = argList.arguments;
+  if (args.length != 2) return null;
+
+  final configPattern = RegExp(r'_?\$\S*Config$');
+
+  final generatedArg = args[1];
+  return generatedArg is SimpleIdentifier &&
+          generatedArg.name.startsWith(configPattern)
+      ? generatedArg
+      : null;
+}
+
+/// Returns whether or not [node] is a class component factory declaration.
+bool isClassComponentFactory(TopLevelVariableDeclaration node) {
+  final type = node.variables?.type;
+  if (type != null && type is NamedType && type?.name?.name == 'UiFactory') {
+    final initializer = node.variables?.variables?.first?.initializer;
+    if (initializer is SimpleIdentifier) {
+      return initializer.name.startsWith('_\$');
+    } else if (initializer is MethodInvocation &&
+        initializer.methodName.name == castFunctionName) {
+      return initializer.argumentList.arguments.first
+          .toSource()
+          .startsWith('_\$');
+    }
+  }
+
+  return false;
+}
+
+/// Returns whether or not [node] is in the legacy boilerplate syntax.
+bool isLegacyFactoryDecl(TopLevelVariableDeclaration node) {
+  final annotation = node.metadata?.firstWhere(
+      (m) => m.toSource().startsWith('@Factory'),
+      orElse: () => null);
+  return isClassComponentFactory(node) && annotation != null;
+}
+
+/// Iterates through the [comment] token stream and removes all ignore
+/// comments containing [ignoreToRemove] using the [yieldPatch] provided.
+///
+/// The entire ignore comment will only be removed if [ignoreToRemove] is the only
+/// item in the list to ignore. Otherwise, [ignoreToRemove] will just be
+/// removed from the list.
+///
+/// Example:
+///
+///   Calling the utility like this:
+///   ```
+///   removeIgnoreComment(
+///     node.beginToken.precedingComments,
+///     'undefined_identifier',
+///     yieldPatch,
+///   );
+///   ```
+///   Will yield the following changes depending on the value of [comment]:
+///
+///   `// ignore: undefined_identifier` => entire comment will be removed
+///
+///   `// ignore: invalid_assignment, undefined_identifier` => `// ignore: invalid_assignment`
+///
+///   `// ignore: invalid_assignment` => `// ignore: invalid_assignment`
+void removeIgnoreComment(
+    Token comment, String ignoreToRemove, YieldPatch yieldPatch) {
+  if (comment == null) return;
+
+  final lexeme = comment.lexeme.replaceAll(' ', '').toLowerCase();
+  if (lexeme.startsWith('//ignore:')) {
+    final ignoreList =
+        lexeme.replaceFirst(RegExp('\/\/ignore\:'), '').split(',');
+    if (ignoreToRemove == null ||
```

Comment:
It looks like the comment is always removed if ignoreToRemove is `null`.

Is that used anywhere? If so, it should be documented. 

If not, perhaps that functionality should be removed and a Argument.notNull error thrown.

Reply:
oh good catch! That definitely should not happen. I'll add `ArgumentError.checkNotNull`